codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -130,7 +130,7 @@ let registerDynamicContracts = (
   } else {
     let newPartition = FetchState.make(
       ~partitionId=partitions->Array.length,
-      ~startBlock,
+      ~startBlock=dynamicContractRegistration.registeringEventBlockNumber,
```

Comment:
This is a fix for the first issue. I had to fix the second one with a huge refactoring to be able to reproduce and find the little first one 😁

Reply:
Ah man, I had a feeling there was something going wrong when creating new partitions. Good catch